libdevcore/CommonData.h

```diff
@@ -42,22 +42,36 @@ enum class WhenError
 	Throw = 1,
 };
 
-enum class HexPrefix
+template <class Iterator>
+std::string toHex(Iterator _it, Iterator _end, std::string _prefix)
 {
-	DontAdd = 0,
-	Add = 1,
-};
-/// Convert a series of bytes to the corresponding string of hex duplets.
-/// @param _w specifies the width of the first of the elements. Defaults to two - enough to represent a byte.
+	typedef std::iterator_traits<Iterator> traits;
+	static_assert(sizeof(typename traits::value_type) == 1, "toHex needs byte-sized element type");
+
+	static char const* hexdigits = "0123456789abcdef";
+	size_t off = _prefix.size();
+	std::string hex(std::distance(_it, _end)*2 + off, '0');
+	hex.replace(0, off, _prefix);
+	for (; _it != _end; _it++)
+	{
+		hex[off++] = hexdigits[*_it >> 4];
```

Comment:
Could you please re-add the conversion to unsigned, just in case we ever use that with a signed type?

`(typename std::make_unsigned<decltype(i)>::type)i`